tests/api/testcl.hpp

```diff
@@ -757,3 +757,20 @@ class WithProfiledCommandQueue : public WithCommandQueue {
 protected:
     void SetUp() override { SetUpQueue(CL_QUEUE_PROFILING_ENABLE); }
 };
+
+class WithPrintfEnabled : public WithCommandQueue {
+protected:
+    void SetUp() override{};
+    void TearDown() override {
+        cl_int err = clReleaseContext(m_context);
+        ASSERT_CL_SUCCESS(err);
+    }
+
+    void SetupPrintfCallback(const cl_context_properties* props) {
+        cl_int err;
+        m_context = clCreateContext(props, 1, &gDevice, nullptr, nullptr, &err);
```

Comment:
Let's keep `clCreateContext` in `WithContext`. Add a `SetUpContext(const cl_context_properties*)` method in `WithContext`.